src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java

```diff
@@ -15,88 +16,81 @@
-    public User saveUser(User User)  {
-
-        String insertQuery = "INSERT INTO login (ID, USERNAME, PASSWORD, ENTITY_ID, DATE_CREATED, DATE_MODIFIED) VALUES (?, ?, ?, ?, ?, ?)";
+    public User saveUser(User user) {
 
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(insertQuery)) {
-            long MaxId = getMaxUserId();
-
-            preparedStatement.setLong(1, MaxId + 1);
-            preparedStatement.setString(2, User.getUsername());
-            preparedStatement.setString(3, User.getPassword());
-            preparedStatement.setString(4, User.getEntity_id());
-            preparedStatement.setTimestamp(5, User.getDate_created());
-            preparedStatement.setTimestamp(6, User.getDate_modified());
+             PreparedStatement preparedStatement = connection.prepareStatement(SAVE_USER_STATEMENT)) {
+            long maxId = getMaxUserId();
+
+            preparedStatement.setLong(1, maxId + 1);
+            preparedStatement.setString(2, user.getUsername());
+            preparedStatement.setString(3, user.getPassword());
+            preparedStatement.setString(4, user.getEntity_id());
+            preparedStatement.setTimestamp(5, user.getDate_created());
+            preparedStatement.setTimestamp(6, user.getDate_modified());
             preparedStatement.executeUpdate();
         } catch (Exception e) {
             LOGGER.error("An SQL Exception occurred." + e.getMessage());
         }
         LOGGER.debug("Saving User failed.");
-        return User;
+        return user;
     }
 
     @Override
-    public long getMaxUserId(){
-
-        String selectMaxIdQuery = "SELECT MAX(ID) AS MAX_ID FROM login";
+    public long getMaxUserId() {
 
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(selectMaxIdQuery);
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_MAX_USER_ID_STATEMENT);
              ResultSet resultSet = preparedStatement.executeQuery()) {
             if (resultSet.next()) {
                 LOGGER.debug("MaxUser ID get successfully.");
                 return resultSet.getLong("MAX_ID");
             }
         } catch (Exception e) {
             LOGGER.error("An SQL Exception occurred." + e.getMessage());
-
         }
         LOGGER.debug("Getting MaxUser ID failed.");
         return 0;
     }
-    @Override
-    public List<User> getAllUsers(){
 
+    @Override
+    public List<User> getAllUsers() {
         List<User> userList = new ArrayList<>();
-        String query = "SELECT * FROM LOGIN";
 
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(query);
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_ALL_USER_STATEMENT);
```

Comment:
fix the try-with-resource. remove preparedstatement from try()

Reply:
okay napo sir